core/src/main/java/org/jdbi/v3/core/mapper/reflect/ConstructorMapper.java

```diff
@@ -75,6 +75,9 @@ public final class ConstructorMapper<T> implements RowMapper<T> {
             + "that your result set has the columns expected, annotate the "
             + "parameter names explicitly with @ColumnName, or annotate nullable parameters as @Nullable";
 
+
+    static final Map<Constructor, Type[]> CONSTRUCTOR_CACHE = new ConcurrentHashMap<>();
```

Comment:
I'll try that out, thanks!

Reply:
updated. Originally I was going to use a guava cache but I see that guava is only imported in test scope. lmk if you think it's a good idea to add a size limit, cache stats, ability to clear cache, disable cache (maybe for benchmarking), etc.

Reply:
We do not depend on Guava from Jdbi core, since it is a huge dep to push onto our users. Additionally, there are numerous known cache bugs that seemingly will never get fixed. The Guava `CacheBuilder` javadocs tell you to use Caffeine instead to avoid such bugs.

I think for a simple use case like this, a synchronized weak hash map is good enough. A map that caches JVM entities like constructors and classes will not grow without bound, so there is no need to size it.